### PR TITLE
Fail 'plural init' if a user is currently running a gcp demo

### PR DESCRIPTION
**Pull request by zreigz**
## Summary
Fail 'plural init' if a user is currently running a GCP demo

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [X] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [ ] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

File: cmd/plural/init.go [hunk 1]
```diff
@@ -21,10 +21,21 @@ import (
...
 
-func handleInit(c *cli.Context) error {
+const DemoingErrorMsg = "You're currently running a gcp demo cluster. Spin that down at https://app.plural.sh/shell before beginning a local installation"
+
+func (p *Plural) handleInit(c *cli.Context) error {
+	p.InitPluralClient()
 	gitCreated := false
 	repo := ""
 
+	me, err := p.Me()
```
Comment (michaeljguarino) [inline cmd/plural/init.go:31]: I'm pretty sure this needs to be done after `handleLogin` (a fresh user won't have a way to query the api before)

**Review by michaeljguarino — CHANGES_REQUESTED** [review]

File: cmd/plural/init.go (hunk 1 above)
Comment (zreigz) [reply cmd/plural/init.go:31]: yes, you are right

**Comment by zreigz** [comment]
@michaeljguarino I have applied your comment. I will add unit and e2e tests later because it requires some refactoring.

File: cmd/plural/init.go [hunk 2]
```diff
@@ -21,7 +21,10 @@ import (
 	"github.com/pluralsh/plural/pkg/wkspace"
 )
 
-func handleInit(c *cli.Context) error {
+const DemoingErrorMsg = "You're currently running a gcp demo cluster. Spin that down at https://app.plural.sh/shell before beginning a local installation"
+
+func (p *Plural) handleInit(c *cli.Context) error {
+	p.InitPluralClient()
```
Comment (michaeljguarino) [inline cmd/plural/init.go:27]: This doesn't have any issues when there's no config created yet?

Comment (zreigz) [reply cmd/plural/init.go]: done

**Review by michaeljguarino — APPROVED** [review]

**Pull request merged**